### PR TITLE
[Enhancement] Make ShowProcStmt get the corresponding forwarding policy based on path

### LINKED ISSUE #9155: Make ShowProcStmt get the corresponding forwarding policy based on path
## Enhancement
Make ShowProcStmt get the corresponding forwarding policy based on path

### PR DESCRIPTION
**Pull request by waittttting**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9155

## Problem Summary(Required) ：

Currently ShowProcStmt supports many paths, but can only support one policy of forwarding master, this PR will Make ShowProcStmt get the corresponding forwarding policy based on path

### CODE REVIEW

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 66 (01.52%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryFragmentProcNode.java|0|1|00.00%|[111]|
|:large_blue_circle:|com/starrocks/common/proc/TablesProcDir.java|0|1|00.00%|[225]|
|:large_blue_circle:|com/starrocks/common/proc/BaseProcNode.java|0|1|00.00%|[51]|
|:large_blue_circle:|com/starrocks/common/proc/ColocationGroupProcDir.java|0|1|00.00%|[81]|
|:large_blue_circle:|com/starrocks/analysis/ShowProcStmt.java|0|1|00.00%|[83]|
|:large_blue_circle:|com/starrocks/common/proc/UserPropertyProcNode.java|0|1|00.00%|[52]|
|:large_blue_circle:|com/starrocks/common/proc/SchedulerWorkingSlotsProcDir.java|0|1|00.00%|[75]|
|:large_blue_circle:|com/starrocks/common/proc/TransStateProcDir.java|0|1|00.00%|[71]|
|:large_blue_circle:|com/starrocks/common/proc/TableProcDir.java|0|1|00.00%|[77]|
|:large_blue_circle:|com/starrocks/common/proc/RollupProcDir.java|0|1|00.00%|[69]|
|:large_blue_circle:|com/starrocks/common/proc/BackendsProcDir.java|0|1|00.00%|[241]|
|:large_blue_circle:|com/starrocks/common/proc/AuthProcDir.java|0|1|00.00%|[77]|
|:large_blue_circle:|com/starrocks/common/proc/IndexSchemaProcNode.java|0|1|00.00%|[92]|
|:large_blue_circle:|com/starrocks/common/proc/TransProcDir.java|0|1|00.00%|[73]|
|:large_blue_circle:|com/starrocks/common/proc/LoadErrorHubProcNode.java|0|1|00.00%|[50]|
|:large_blue_circle:|com/starrocks/common/proc/IndexInfoProcDir.java|0|1|00.00%|[110]|
|:large_blue_circle:|com/starrocks/common/proc/HMSTablePartitionsProcDir.java|0|1|00.00%|[70]|
|:large_blue_circle:|com/starrocks/common/proc/ClusterLoadStatByMedium.java|0|1|00.00%|[57]|
|:large_blue_circle:|com/starrocks/common/proc/FrontendsProcNode.java|0|1|00.00%|[162]|
|:large_blue_circle:|com/starrocks/common/proc/JvmProcDir.java|0|1|00.00%|[110]|
|:large_blue_circle:|com/starrocks/common/proc/EsShardProcDir.java|0|1|00.00%|[113]|
|:large_blue_circle:|com/starrocks/common/proc/TaskFailedProcNode.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/common/proc/LoadProcDir.java|0|1|00.00%|[114]|
|:large_blue_circle:|com/starrocks/common/proc/EsPartitionsProcDir.java|0|1|00.00%|[141]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQuerySqlProcDir.java|0|1|00.00%|[77]|
|:large_blue_circle:|com/starrocks/common/proc/DbsProcDir.java|0|1|00.00%|[143]|
|:large_blue_circle:|com/starrocks/common/proc/BackendProcNode.java|0|1|00.00%|[134]|
|:large_blue_circle:|com/starrocks/common/proc/TaskTypeProcNode.java|0|1|00.00%|[96]|
|:large_blue_circle:|com/starrocks/common/proc/LocalTabletsProcDir.java|0|1|00.00%|[231]|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|0|1|00.00%|[244]|
|:large_blue_circle:|com/starrocks/common/proc/RollupJobProcNode.java|0|1|00.00%|[54]|
|:large_blue_circle:|com/starrocks/common/proc/ReplicasProcNode.java|0|1|00.00%|[109]|
|:large_blue_circle:|com/starrocks/common/proc/LakeTabletsProcNode.java|0|1|00.00%|[89]|
|:large_blue_circle:|com/starrocks/common/proc/ExportProcNode.java|0|1|00.00%|[77]|
|:large_blue_circle:|com/starrocks/common/proc/PartitionsProcDir.java|0|1|00.00%|[380]|
|:large_blue_circle:|com/starrocks/common/proc/TasksProcDir.java|0|1|00.00%|[89]|
|:large_blue_circle:|com/starrocks/common/proc/StatisticProcDir.java|0|1|00.00%|[245]|
|:large_blue_circle:|com/starrocks/common/proc/IndicesProcDir.java|0|1|00.00%|[139]|
|:large_blue_circle:|com/starrocks/common/proc/ClusterBalanceProcDir.java|0|1|00.00%|[90]|
|:large_blue_circle:|com/starrocks/common/proc/ClusterLoadStatisticProcDir.java|0|1|00.00%|[80]|
|:large_blue_circle:|com/starrocks/common/proc/ColocationGroupBackendSeqsProcNode.java|0|1|00.00%|[58]|
|:large_blue_circle:|com/starrocks/common/proc/RoutineLoadProcNode.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/common/proc/RoutineLoadsProcDir.java|0|1|00.00%|[84]|
|:large_blue_circle:|com/starrocks/common/proc/PriorityRepairProcNode.java|0|1|00.00%|[52]|
|:large_blue_circle:|com/starrocks/common/proc/IncompleteTabletsProcNode.java|0|1|00.00%|[70]|
|:large_blue_circle:|com/starrocks/common/proc/ComputeNodeProcDir.java|0|1|00.00%|[142]|
|:large_blue_circle:|com/starrocks/common/proc/TransTablesProcDir.java|0|1|00.00%|[81]|
|:large_blue_circle:|com/starrocks/common/proc/JobsProcDir.java|0|1|00.00%|[149]|
|:large_blue_circle:|com/starrocks/common/proc/BaseProcDir.java|0|1|00.00%|[59]|
|:large_blue_circle:|com/starrocks/common/proc/SchemaChangeJobProcNode.java|0|1|00.00%|[54]|
|:large_blue_circle:|com/starrocks/catalog/ResourceMgr.java|0|1|00.00%|[272]|
|:large_blue_circle:|com/starrocks/common/proc/DeleteInfoProcDir.java|0|1|00.00%|[68]|
|:large_blue_circle:|com/starrocks/catalog/BrokerMgr.java|0|1|00.00%|[361]|
|:large_blue_circle:|com/starrocks/common/proc/MonitorProcDir.java|0|1|00.00%|[70]|
|:large_blue_circle:|com/starrocks/common/proc/TransDbProcDir.java|0|1|00.00%|[67]|
|:large_blue_circle:|com/starrocks/common/proc/TabletSchedulerDetailProcDir.java|0|1|00.00%|[79]|
|:large_blue_circle:|com/starrocks/common/proc/SchedulerStatProcNode.java|0|1|00.00%|[53]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryBackendInstanceProcDir.java|0|1|00.00%|[58]|
|:large_blue_circle:|com/starrocks/common/proc/TransPartitionProcNode.java|0|1|00.00%|[70]|
|:large_blue_circle:|com/starrocks/common/proc/JobsDbProcDir.java|0|1|00.00%|[98]|
|:large_blue_circle:|com/starrocks/common/proc/SchemaChangeProcDir.java|0|1|00.00%|[182]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryStatisticsProcDir.java|0|1|00.00%|[109]|
|:large_blue_circle:|com/starrocks/common/proc/RoutineLoadsNameProcDir.java|0|1|00.00%|[85]|
|:large_blue_circle:|com/starrocks/common/proc/LoadJobProcNode.java|0|1|00.00%|[46]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|2|50.00%|[336]|

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 9 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/ShowProcStmt.java|6|9|66.67%|[96, 97, 98]|

**Review by gengjun-git — DISMISSED** [review]

**Review by padmejin — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/ShowProcStmt.java|9|9|100.00%|[]|

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 11 / 11 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/ShowProcStmt.java|11|11|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/analysis/ShowProcStmt.java [hunk 1]
```diff
@@ -32,6 +35,16 @@
 
 // SHOW PROC statement. Used to show proc information, only admin can use.
 public class ShowProcStmt extends ShowStmt {
+
+    public static final Set<String> needForwardPathRoot;
+    static {
+        needForwardPathRoot = new HashSet<String>();
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/analysis/ShowProcStmt.java:41]: better use ImmutableSet.of

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/analysis/ShowProcStmt.java (hunk 1 above)
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/analysis/ShowProcStmt.java]: done

**Review by Astralidea — DISMISSED** [review]

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 12 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/ShowProcStmt.java|12|12|100.00%|[]|

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 12 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/ShowProcStmt.java|12|12|100.00%|[]|

**Review by Astralidea — APPROVED** [review]

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by waittttting** [comment]
run starrocks_clang-format

**Comment by waittttting** [comment]
run starrocks_clang-format

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**